spec/support/concerns/restorable_shared_examples.rb

```diff
@@ -135,84 +141,87 @@
   end #manage_children
 
   describe '#restore_children' do
-    it { is_expected.not_to respond_to(:restore_children).with(0).arguments }
-    it { is_expected.to respond_to(:restore_children).with(1).argument }
-
-    context 'called', :vcr do
-      include_context 'with job runner', ChildRestorationJob
-      let(:job_transaction) { ChildRestorationJob.initialize_job(subject) }
-      let(:child_job_transaction) { ChildRestorationJob.initialize_job(child_folder) }
-      let(:child_folder_file) { FactoryGirl.create(:data_file, parent: child_folder)}
-      let(:page) { 1 }
-
-      before do
-        expect(child_folder).to be_persisted
-        child_folder.update_column(:is_deleted, true)
-        expect(child_folder_file).to be_persisted
-        child_folder_file.update_column(:is_deleted, true)
-        valid_child_file.update_column(:is_deleted, true)
-        expect(valid_child_file.is_deleted?).to be_truthy
-
-        @old_max = Rails.application.config.max_children_per_job
-        Rails.application.config.max_children_per_job = subject.children.count + child_folder.children.count
-      end
+    include_context 'with job runner', ChildRestorationJob
+    let(:job_transaction) { ChildRestorationJob.initialize_job(subject) }
+    if child_minder_children_sym
+      let(:child_job_transaction) { ChildRestorationJob.initialize_job(child_minder_children.first) }
+    end
+
+    let(:page) { 1 }
 
-      after do
-        Rails.application.config.max_children_per_job = @old_max
+    before do
+      expected_children.each do |expected_child|
```

Comment:
It would be good to check that `expected_children` is not empty before calling `#each` on it. If it's empty, a lot of the expect statements in this section will be silently skipped. There are probably other places that could benefit from this sort of check.